src/factory-plugin.ts

```diff
@@ -0,0 +1,27 @@
+/*********************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import * as theia from '@theia/plugin';
+import { FactoryProjectsManager } from "./factory-projects-manager";

```

Comment:
I don't like the name `...ProjectsManager` because it is a lot more then just managing projects,

Reply:
FactoryHandler?

Reply:
actually I don't like `FactoryProjectsManager` also

Reply:
Decided to rename it to `FactoryInitializer`